dvc/stage/cache.py

```diff
@@ -84,7 +124,13 @@ def save(self, stage):
         cache = to_single_stage_lockfile(stage)
         cache_value = _get_cache_hash(cache)
 
-        if self._load_cache(cache_key, cache_value):
+        existing_cache = self._load_cache(cache_key, cache_value)
+        cache = existing_cache or cache
+
+        for out in self._uncached_outs(stage, cache):
+            out.commit()
```

Comment:
`outs_no_cache` is more or less `do-not-push`, right? "outputs are not tracked by DVC" is no longer the case.

Reply:
Also, if the stage cache can be shared, but `outs_no_cache` are not being shared at all. Does this make sense at all?

Reply:
Yes, we forcefully cache and share `outs_no_cache` if you use build cache. E.g. if you have a git-tracked metrics for your stage, run-cache will replace it to what it should be.

Reply:
Got it. I was confused that when I did `dvc push`, non-cached outputs was not on remote. I had forgotten that it requires `--run-cache` flag.